pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/KitRequestShipping.java

```diff
@@ -197,6 +197,13 @@ public class KitRequestShipping extends KitRequest implements HasDdpInstanceId {
                     + "denial_reason = ?, authorized_by = ? WHERE kit.dsm_kit_request_id = ?";
     private static final String MARK_ORDER_AS_TRANSMITTED =
             "update ddp_kit_request set order_transmitted_at = ? where external_order_number = ?";
+
+    private static final String GET_COUNT_KITS_WITH_SAME_COLLABORATOR_SAMPLE_ID_AND_KIT_TYPE =
+            " SELECT count(*) kitRequestCount from ddp_kit_request where bsp_collaborator_sample_id REGEXP \"^%1\" and kit_type_id = ?";
```

Comment:
I think you mean collaborator sample ids, they are made like `<project name>_<short id>_<sample type>_<counter>`.  SM IDs are barcodes assigned to tissue samples by study staff or GP, I _think_ they are made like `SM-<some number>`

Reply:
So here we check for if there are any samples with collaborator sample id starting with `<project name>_<short id>_<sample type>` to see what the counter should be

Reply:
But is the replacement for `%1` actually a regular expression? If not we should get rid of `REGEXP` in the query since it can only cause subtle bugs if a reg expression character ever appears in a collaborator sample ID.

Reply:
Yes it starts with `^`